icu4j/main/classes/core/src/com/ibm/icu/message2/Mf2ParseException.java

```diff
@@ -0,0 +1,73 @@
+// © 2022 and later: Unicode, Inc. and others.
+// License & terms of use: http://www.unicode.org/copyright.html
+
+package com.ibm.icu.message2;
+
+/**
+ * An exception thrown when parsing a {@link MessageFormatter} syntax fails.
+ *
+ * @internal ICU 72 technology preview
+ * @deprecated This API is for ICU internal use only.
+ */
+@Deprecated
+public class Mf2ParseException extends RuntimeException {
```

Comment:
Discuss: Should this be a different exception class from MessageFormatterException or IllegalArgumentException?

Reply:
I checked what MessageFormat does: `java.lang.IllegalArgumentException`
I can do that, and remove `Mf2ParseException`.
Your call, I don't mind.

Reply:
Done.
Replaced all instances with `IllegalArgumentException` and removed `Mf2ParseException`